ui/src/lib/DarkMode.svelte

```diff
@@ -0,0 +1,178 @@
+<script>
+  import { onMount } from 'svelte';
+  import logo from '../assets/images/songstitch_logo.png';
+  import logoDark from '../assets/images/songstitch_logo_dark.png';
+  import darkModeIcon from '../assets/images/dark_mode.png';
+  let darkMode = localStorage.getItem('darkMode') === 'true';
+
+  function toggleClassByClassname(className) {
+    var elements = document.getElementsByClassName(className);
+    for (var i = 0; i < elements.length; i++) {
+      elements[i].classList.toggle('dark-mode');
+    }
+  }
+
+  function applyDarkMode() {
+    document.body.classList.toggle('dark-mode');
+    document.querySelector('html').classList.toggle('dark-mode');
+    document.getElementsByTagName('dialog')[0].classList.toggle('dark-mode');
+    document.getElementsByTagName('fieldset')[0].classList.toggle('dark-mode');
+    document.querySelector('#app > main > form').classList.toggle('dark-mode');
+
+    const selectElements = document.getElementsByTagName('select');
```

Comment:
We should be able to make this a lot simpler by using something like what was done here https://svelte.dev/repl/ed4fef4beceb4b0eb295d1f9fdf3bd62?version=3.6.9